nitro-node/src/index.ts

```diff
@@ -0,0 +1,277 @@
+/**
+ * @file This file exports a class that implements the InferenceExtension interface from the @janhq/core package.
+ * The class provides methods for initializing and stopping a model, and for making inference requests.
+ * It also subscribes to events emitted by the @janhq/core package and handles new message requests.
+ * @version 1.0.0
+ * @module inference-extension/src/index
+ */
+
+import {
+  ChatCompletionRole,
+  ContentType,
+  MessageRequest,
+  MessageStatus,
+  ThreadContent,
+  ThreadMessage,
+  events,
+  executeOnMain,
+  fs,
+  Model,
+  joinPath,
+  InferenceExtension,
+  log,
+  InferenceEngine,
+  MessageEvent,
+  ModelEvent,
+  InferenceEvent,
+} from "@janhq/core";
```

Comment:
I see that you moved the entire nitro extension source code into this package. This would couple the Jan app business logic, which is not the idea of the nitro-node, as a stand-alone module.

1. How do you define these logics? For example, how do they work outside of the Jan app (other node apps)?
2. Is this a browser module or a Node.js module?

Reply:
Noted, after the build workflow for both Jan and nitro-node can be run independently as of now I'm beginning to refactor the source code inside nitro-node. Should take some time before the PR is ready...🧐